onnxruntime/core/graph/graph_flatbuffers_utils.cc

```diff
@@ -344,12 +462,15 @@ Status LoadAttributeOrtFormat(const fbs::Attribute& fbs_attr,
 Status SaveOrtTensorOrtFormat(
     const std::string& tensor_name, const onnxruntime::Tensor& ort_tensor,
     flatbuffers::FlatBufferBuilder& builder,
-    flatbuffers::Offset<fbs::Tensor>& fbs_tensor) {
+    flatbuffers::Offset<fbs::Tensor>& fbs_tensor,
+    ExternalDataWriter external_data_writer) {
   ORT_RETURN_IF(ort_tensor.IsDataTypeString(),
                 "TensorProto_DataType_STRING is not supported while saving a tensor to ORT format.");
 
   const auto fbs_tensor_name = builder.CreateString(tensor_name);
   const auto fbs_tensor_dims = SaveDims(builder, ort_tensor.Shape().GetDims());
+  // To avoid issues with vtable offsets, raw_data fbs::vector must be constructed before the TensorBuilder begins
+  // building the tensor. See flatbuffer_builder.h's NotNested() function for more details.
   flatbuffers::Offset<flatbuffers::Vector<uint8_t>> raw_data = builder.CreateVector(
```

Comment:
Ah, yes, it's a complication with how the FlatBuffer Builder works -- the function will error out otherwise in one of the flatbuffer_builder functions.

Reply:
Can we do this so we only populate if we're going to use it?
 
```
  flatbuffers::Offset<flatbuffers::Vector<uint8_t>> raw_data;
  if (!external_data_writer) {
    raw_data = builder.CreateVector(static_cast<const uint8_t*>(ort_tensor.DataRaw()),
                                    ort_tensor.SizeInBytes());
  }
```

Reply:
oh wait yea that would work